flake.nix

```diff
@@ -54,6 +55,13 @@
               homepage = "https://git.peppe.rs/languages/statix/about";
               license = licenses.mit;
             };
+
+            nativeCheckInputs = [ clippy ];
+
+            checkPhase = ''
+              cargo clippy
+              cargo test
+            '';
```

Comment:
Perhaps [`postCheck`](https://nixos.org/manual/nixpkgs/unstable/#var-stdenv-postCheck), to avoid overriding the default `checkPhase`?

Reply:
Good idea. Done via https://github.com/oppiliappan/statix/pull/94/commits/82c4576c4ddaaa062449cdd4154b273cadb9ebf5